diff -r c7f419836544 authinfo.py
--- a/authinfo.py	Wed May 19 12:31:41 2010 +0200
+++ b/authinfo.py	Fri May 21 16:57:37 2010 +0200
@@ -3449,10 +3449,11 @@
 		return True
 
 	def prewriteUpdate(self):
-		self.implicitSSSD = self.implicitSSSDAuth = self.sssdSupported()
-		if self.implicitSSSD:
-			# we force the update if in the pristine copy it was set to True
-			self.forceSSSDUpdate = False
+		if not self.enableSSSD and not self.enableSSSDAuth:
+			self.implicitSSSD = self.implicitSSSDAuth = self.sssdSupported()
+			if self.implicitSSSD:
+				# we force the update if in the pristine copy it was set to True
+				self.forceSSSDUpdate = False
 		modules = getSmartcardModules()
 		if len(modules) > 0 and self.smartcardModule not in modules:
 			self.smartcardModule = modules[0]
@@ -3482,6 +3483,8 @@
 				ret = ret and self.writeSMB()
 			if self.enableWinbind or self.enableWinbindAuth:
 				ret = ret and self.writeWinbind()
+			if self.implicitSSSD or self.implicitSSSDAuth:
+				ret = ret and self.writeSSSD()
 			ret = ret and self.writeNSS()
 			ret = ret and self.writePAM()
 			ret = ret and self.writeSysconfig()
