diff -r 0f3b95c86b85 authconfig.py
--- a/authconfig.py	Tue Jun 22 16:45:38 2010 +0000
+++ b/authconfig.py	Mon Jul 12 16:01:28 2010 +0200
@@ -522,8 +522,8 @@
 		warnCache = [authinfo.PATH_NSCD, _("caching"), "nscd", None]
 		warnFprintd = [authinfo.PATH_PAM_FPRINTD, _("Fingerprint reader"), "pam_fprintd", None]
 		warnKerberos = [authinfo.PATH_PAM_KRB5, _("Kerberos"), "pam_krb5", None]
-		warnLDAPAuth = [authinfo.PATH_PAM_LDAP, _("LDAP authentication"), "nss_ldap", None]
-		warnLDAP = [authinfo.PATH_LIBNSS_LDAP, _("LDAP"), "nss_ldap", None]
+		warnLDAPAuth = [authinfo.PATH_PAM_LDAP, _("LDAP authentication"), "pam_ldap", None]
+		warnLDAP = [authinfo.PATH_LIBNSS_LDAP, _("LDAP"), "nss-pam-ldapd", None]
 		warnNIS = [authinfo.PATH_YPBIND, _("NIS"), "ypbind", None]
 		warnShadow = [authinfo.PATH_PWCONV, _("shadow password"), "shadow-utils", None]
 		warnWinbindNet = [authinfo.PATH_WINBIND_NET, _("Winbind"), "samba-client", None]
